lib/rouge/lexers/kotlin.rb

```diff
@@ -85,7 +73,13 @@ class Kotlin < RegexLexer
         rule %r'"(\\\\|\\"|[^"\n])*["\n]'m, Str
         rule %r"'\\.'|'[^\\]'", Str::Char
         rule %r"[0-9](\.[0-9]+)?([eE][+-][0-9]+)?[flFL]?|0[xX][0-9a-fA-F]+[Ll]?", Num
-        rule %r/@#{id}/, Name::Decorator
+        rule %r/(@#{upper}#{name_backtick})/, Name::Decorator
+        rule %r'(#{upper}#{name_backtick})(<)' do
+          groups Name::Class, Punctuation
+          push :generic_parameters
+        end
+        rule %r'(#{upper}#{name_backtick})', Name::Class
+        rule %r'(#{name_backtick})(?=\s*[({])', Name::Function
```

Comment:
@lordcodes I realise in one lexer an additional regular expression doesn't matter but we try to reduce the memory necessary for object allocation as much as possible. To that end, the lowercase regex didn't seem necessary so long as the rules were differently organised.

Reply:
Oh, and the visual sample looked fine to me after this change. Let me know what you think.

Reply:
Ah good point, looks like a good change to me!